e2e/fixtures/status.go

```diff
@@ -260,20 +260,29 @@ func (fdbCluster FdbCluster) StatusInvariantChecker(
 	)
 }
 
+// checkAvailability returns nil if the cluster is reachable or if not a quorum of the coordinators are reachable. If
+// the cluster is unreachable an error will be returned.
+func checkAvailability(status *fdbv1beta2.FoundationDBStatus) error {
+	// If we are not able to reach the quorum of coordinators, we cannot make any assumption about the status of the Cluster.
+	if !status.Client.Coordinators.QuorumReachable {
+		return nil
+	}
+
+	if !status.Client.DatabaseStatus.Available {
```

Comment:
Sorry, I am missing the context here - if a quorum of coordinators is not reachable then shouldn't the status be empty and so this flag should have been cleared? Maybe I need to check the logic used for setting/clearing these flags.

Reply:
The problem here is that this value is not a pointer. So if the value is not set and we convert it to the `FoundationDBStatus` it will be `false` (default) value. Since this is an exposed "API" or struct we cannot just change it as it would break things. I have to verify if we can have the json struct point to two fields.

Reply:
> So if the value is not set and we convert it to the FoundationDBStatus it will be false (default) value.

The default value should work here, isn't it? Are you saying "Available" gets set to "true" if a quorum of coordinators is not reachable?

Reply:
Updated the code.